Source/Filters/Trim/YPTimeStampScrollView.swift

```diff
@@ -31,41 +48,83 @@ class YPTimeStampScrollableView: UIScrollView {
+            // Clips greater then 61 seconds will have a scale of 1 dot = 10 seconds and every 3rd dot will render a timestmp label.
+            timeStamps = generateTimeStampViewModels(amount: Int(timeRangeAmount), timeStampInterval: 3, duration: 10)
+        }
+        return timeStamps
+    }
+
+    func renderRanges() {
+        guard let videoDuration = asset?.duration else { return }
 
-        var viewCount = Int(contentSize.width) / width
-        // The viewCount is adjusted here to account for the sliding trim handle views. These views consume horizontal space on the
-        // left and the right sides of the trimmer view, and the contentSize of the trimmer view does not extend before the left sliding view or extend
-        // beyond the right sliding view.
-        viewCount += 2
+        let ranges = generateTimeRanges(for: videoDuration)
+        let subViewWidth = contentSize.width / CGFloat(ranges.count)
+        let handleBarWidth: CGFloat = 15
 
-        for i in 0..<viewCount{
-            let contentViewSubView = YPTrimmerTimeStampView(
+        for i in 0..<ranges.count {
+            let range = ranges[i]
+            let contentViewSubView = YPTimeStampView(
                 timeStampFont: timeStampFont,
                 timeStampColor: timeStampColor,
-                timeBarColor: timeBarColor
+                timeBarColor: range.shouldRenderTimeStamp ? timeBarLargeCircleColor : timeBarSmallCircleColor
             )
 
-            let xPosition = i * width
-
-            if i == 0, let asset = asset {
-                contentViewSubView.shouldRenderBoldCircle = true
-                contentViewSubView.timeStampText = getTime(from: CGFloat(xPosition), for: asset)?.durationText
-            }
+            guard var xPosition = getPosition(from: range.startTime) else { continue }
 
-            // On every 3rd view we want to draw a bold circle.
-            if i % 3 == 0 {
+            if range.shouldRenderTimeStamp, let asset = asset {
                 contentViewSubView.shouldRenderBoldCircle = true
+                contentViewSubView.timeStampText = range.durationText
             }
-
-            // On every 6th view (not counting the starting view) we want to render the video timestamp at that point.
-            if i % 6 == 0, let asset = asset {
-                contentViewSubView.timeStampText = getTime(from: CGFloat(xPosition), for: asset)?.durationText
-            }
-
-            contentViewSubView.frame = CGRect(x: (i * width) + startOffset, y: 0, width: width, height: 23)
+            // We want the first 0:00 dot to line up just after the left handle bar. This offset is to account for the handle bar width and
+            // start the first timestamp just to the right of the left handle.
+            
+            //We also need to move the xPosition forward by the same offset value used to pull the time stamp UI back to the left. This pullback is
```

Comment:
super nit: extra space needed before "We" for conformity.

Reply:
@will-ltk Updated this too.